### DISCUSSION TITLE
Using the flatten filter in scripts and rest sensors causes them to break after a restart

### DISCUSSION THREAD

**Discussion by pbaylas** [frenck/spook#824] [Q&A] [answered]
I was very excited when I discovered Spook because I have so many times found myself reaching for a flatten filter that didn't exist when writing HA templates. I've started using it in various places and run into problems when I restart home assistant.

I tried using it in a restful sensor's `value_template` but I get an error when HA starts because it cannot find the `flatten` filter which then disables the sensor.

Similarly, I am using it in a script, but it also gets disabled after a restart because the filter isn't recognised. I can workaround this by tweaking the script and saving it, but it's annoying to have to do that after every restart.
<img width="622" alt="image" src="https://github.com/user-attachments/assets/74b3054c-0738-4c9a-8c90-549c4f48cca8">

Is there either a way to force spook to load before these other integrations, or to change the templates to fail more gracefully when flatten hasn't loaded yet?

Thanks in advance for any suggestions!

**Comment by pbaylas** [comment]
After writing the above, I discovered the `is filter` test (https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-tests.filter) which seemed to do what I want. However, despite it working as I'd expect in the dev tools, when I use it in a rest sensor, it seems to always fail:

```yaml
        value_template:  >-
          {% if "flatten" is filter %}
            {{ value_json.data.category_groups | map(attribute="categories") | flatten(levels=1) | selectattr("balance", "lt", 0) | list | length }}'
          {% else %}
            unavailable
          {% endif %}
```
Even well after a restart, this sensor is stuck as "unavailable".

**Comment by SLaks** [comment]
I solved this issue by writing an automation that reloads template entities and automations a minute or 2 after HA restarts

**Comment by frenck** [comment, marked as answer]
This issue has been resolved, these features have been directly moved into the Home Assistant Core.

../Frenck
